### PR TITLE
fix: dashboard collaboration is required to change dashboard description

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

Dashboard Collaboration is required in order to change the description for a dashboard in the UI. But not enforced in the API. So an "el33t hack0r" can change the description.

l33t

## Changes

checks permissions while validating the dashboard

## How did you test this code?

adding developer tests. run `./bin/tests ee/api/test/test_dashboard.py`
running it locally and seeing it isn't possible anymore

### CODE REVIEW

File: posthog/api/dashboard.py [hunk 1]
```diff
@@ -61,6 +62,13 @@ class Meta:
         ]
         read_only_fields = ["creation_mode", "effective_restriction_level", "is_shared"]
 
+    def validate_description(self, value: str) -> str:
+        team: Team = Team.objects.get(id=self.context["team_id"])
+        if value and AvailableFeature.DASHBOARD_COLLABORATION not in team.organization.available_features:
+            raise PermissionDenied("You must have paid for dashboard collaboration to set the dashboard description")
+
+        return value
```
Comment (mariusandra) [inline posthog/api/dashboard.py:70]: Will this add N+1 team queries if requesting e.g. a list of dashboards? We do query `https://app.posthog.com/api/projects/$our_team/dashboards/?limit=300` whenever you open the app.

**Review by mariusandra — COMMENTED** [review]
Looks great, but wondering if django uses some magic to cache queries... 🤔

File: posthog/api/dashboard.py (hunk 1 above)
Comment (pauldambra) [reply posthog/api/dashboard.py:70]: Interestingly there must be magic and it's already nplus1. Adding a query to pre-load the team doesn't reduce the number of queries. In. fact there's one _more_ query if I pre-query for the team 🤷‍♂️

Added a test to snapshot and it's the sharing configuration that adds 1 query per dashboard.

Comment (pauldambra) [reply posthog/api/dashboard.py]: Django is weird. Found and fixed the source of the nplus1...

**Review by mariusandra — APPROVED** [review]
Joy was sparked ✨

**Pull request merged**